vscode/src/debugger/activate.ts

```diff
@@ -106,8 +107,26 @@ class QsDebugConfigProvider implements vscode.DebugConfigurationProvider {
         config.request = "launch";
         config.program = editor.document.uri.toString();
         config.shots = 1;
-        config.stopOnEntry = true;
         config.noDebug = "noDebug" in config ? config.noDebug : false;
+        config.stopOnEntry = !config.noDebug;
+      }
+    } else {
+      // we have a launch config, resolve the program path
+
+      // ensure we have the program uri correctly formatted
+      // this is a user specified path.
+      if (config.program && config.program !== "${file}") {
+        const uri = workspaceFileAccessor.resolvePathToUri(config.program);
+        config.program = uri.toString();
+      } else {
+        // Use the active editor if no program or ${file} is specified.
+        const editor = vscode.window.activeTextEditor;
+        if (
+          editor &&
+          vscode.languages.match(qsharpDocumentFilter, editor.document)
+        ) {
+          config.program = editor.document.uri.toString();
+        }
```

Comment:
Do we also want implement `provideDebugConfiguration` so that we can recommend a default configuration to the user? I'm not sure how that would work exactly, perhaps we would need discover .qs files in the workspace? I'm not sure if it's worth it.

Reply:
Not sure, we can look at it in the future.